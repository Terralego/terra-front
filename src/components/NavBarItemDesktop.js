import React from 'react';
import PropTypes from 'prop-types';

import {
  Position,
  Tooltip,
} from '@blueprintjs/core';

import HeaderLink from './HeaderLink';
import HeaderButton from './HeaderButton';

export const NavBarItemDesktop = ({
  id,
  content,
  label,
  href,
  onClick,
  icon,
}) => (
  <Tooltip
    content={label}
    position={Position.RIGHT}
    usePortal={false}
  >
    <HeaderLink
      href={href}
      onClick={onClick}
      data-link-id={id}
    >
      <HeaderButton
        id={id}
        icon={icon}
        alt={label}
      >
        {content}
      </HeaderButton>
    </HeaderLink>
  </Tooltip>
);

NavBarItemDesktop.propTypes = {
  id: PropTypes.string.isRequired,
  href: PropTypes.string.isRequired,
  icon: PropTypes.string,
  label: PropTypes.string,
  content: PropTypes.string,
  onClick: PropTypes.func,
};

NavBarItemDesktop.defaultProps = {
  icon: '',
  label: '',
  content: '',
  onClick () {},
};

export default NavBarItemDesktop;
