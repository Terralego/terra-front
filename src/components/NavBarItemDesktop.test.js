import React from 'react';
import renderer from 'react-test-renderer';
import { NavBarItemDesktop } from './NavBarItemDesktop';

jest.mock('./HeaderLink', () => () => 'HeaderLink');
jest.mock('./HeaderButton', () => () => 'HeaderButton');

it('should render correctly the component', () => {
  const tree = renderer.create(
    <NavBarItemDesktop href="visualiser" id="test" />,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
