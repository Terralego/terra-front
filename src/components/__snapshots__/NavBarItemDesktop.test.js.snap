// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly the component 1`] = `
<span
  className="bp3-popover-wrapper"
>
  <span
    className="bp3-popover-target"
    onBlur={[Function]}
    onFocus={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    HeaderLink
  </span>
</span>
`;
